Parse/PFObject.m

```diff
@@ -2317,6 +2317,36 @@ - (void)removeObjectForKey:(NSString *)key {
     }
 }
 
+- (void)revert {
+    @synchronized (lock) {
+        if ([self isDirty]) {
+            PFOperationSet *changes = [self unsavedChanges];
+            [changes enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
+                [_availableKeys removeObject:key];
+            }];
+            
+            [operationSetQueue removeLastObject];
```

Comment:
This can potential break a lot of things, since we rely on the fact that there is going to be a single `PFOperationSet` in the queue. `PFOperationSet` doesn't have `removeAllObjects` right now, but it would be easy to add.